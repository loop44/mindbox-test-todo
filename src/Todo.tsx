import ArrowDown from './assets/arrow-down.svg';
import Marker from './assets/marker.svg';
import MarkerDone from './assets/marker-done.svg';
import { Form, Todo, TodoInner, TodoPanel, TodoWrapper } from './Todo.elements';

const DEFAULT_TODO_LIST = [
  { id: 1, text: 'Тестовое задание', done: false },
  { id: 2, text: 'Прекрасный код', done: true },
  { id: 3, text: 'Покрытие тестами', done: false }
];

const App = () => (
  <TodoWrapper>
    <h1>todos</h1>
    <TodoInner>
      <Form>
        <input type="image" src={ArrowDown} alt="" />
        <input type="text" placeholder="What needs to be done?" />
      </Form>
      <div>
        {DEFAULT_TODO_LIST.map(({ id, text, done }) => (
          <Todo key={id} done={done}>
            {done ? (
              <input type="image" src={MarkerDone} alt="" />
            ) : (
              <input type="image" src={Marker} alt="" />
            )}
            <span>{text}</span>
          </Todo>
        ))}
      </div>
      <TodoPanel>
        <p>2 items left</p>
        <div className="todoTypes">
          <button className="active">All</button>
          <button>Active</button>
          <button>Completed</button>
        </div>
        <button>Clear completed</button>
      </TodoPanel>
    </TodoInner>
  </TodoWrapper>
);

export default App;
