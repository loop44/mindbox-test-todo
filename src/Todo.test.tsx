import { describe, expect, it } from 'vitest';

import { render, screen } from './utils/test-utils';
import Todo from './Todo';

describe('Todo test', () => {
  it('the title is visible', () => {
    render(<Todo />);
    expect(screen.getByText('todos')).toBeInTheDocument();
  });

  it('todo snapshot', () => {
    const todo = render(<Todo />);

    expect(todo).toMatchSnapshot();
  });
});
