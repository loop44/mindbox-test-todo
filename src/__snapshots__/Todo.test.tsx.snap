// Vitest Snapshot v1

exports[`Todo test > todo snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-bcXHqe kRHNFb"
      >
        <h1>
          todos
        </h1>
        <div
          class="sc-gswNZR gfYYrO"
        >
          <form
            class="sc-dkrFOg isoTNN"
          >
            <input
              alt=""
              src="/src/assets/arrow-down.svg"
              type="image"
            />
            <input
              placeholder="What needs to be done?"
              type="text"
            />
          </form>
          <div>
            <div
              class="sc-hLBbgP fsfPHN"
            >
              <input
                alt=""
                src="/src/assets/marker.svg"
                type="image"
              />
              <span>
                Тестовое задание
              </span>
            </div>
            <div
              class="sc-hLBbgP gHtJIq"
            >
              <input
                alt=""
                src="/src/assets/marker-done.svg"
                type="image"
              />
              <span>
                Прекрасный код
              </span>
            </div>
            <div
              class="sc-hLBbgP fsfPHN"
            >
              <input
                alt=""
                src="/src/assets/marker.svg"
                type="image"
              />
              <span>
                Покрытие тестами
              </span>
            </div>
          </div>
          <div
            class="sc-eDvSVe kziEai"
          >
            <p>
              2 items left
            </p>
            <div
              class="todoTypes"
            >
              <button
                class="active"
              >
                All
              </button>
              <button>
                Active
              </button>
              <button>
                Completed
              </button>
            </div>
            <button>
              Clear completed
            </button>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-bcXHqe kRHNFb"
    >
      <h1>
        todos
      </h1>
      <div
        class="sc-gswNZR gfYYrO"
      >
        <form
          class="sc-dkrFOg isoTNN"
        >
          <input
            alt=""
            src="/src/assets/arrow-down.svg"
            type="image"
          />
          <input
            placeholder="What needs to be done?"
            type="text"
          />
        </form>
        <div>
          <div
            class="sc-hLBbgP fsfPHN"
          >
            <input
              alt=""
              src="/src/assets/marker.svg"
              type="image"
            />
            <span>
              Тестовое задание
            </span>
          </div>
          <div
            class="sc-hLBbgP gHtJIq"
          >
            <input
              alt=""
              src="/src/assets/marker-done.svg"
              type="image"
            />
            <span>
              Прекрасный код
            </span>
          </div>
          <div
            class="sc-hLBbgP fsfPHN"
          >
            <input
              alt=""
              src="/src/assets/marker.svg"
              type="image"
            />
            <span>
              Покрытие тестами
            </span>
          </div>
        </div>
        <div
          class="sc-eDvSVe kziEai"
        >
          <p>
            2 items left
          </p>
          <div
            class="todoTypes"
          >
            <button
              class="active"
            >
              All
            </button>
            <button>
              Active
            </button>
            <button>
              Completed
            </button>
          </div>
          <button>
            Clear completed
          </button>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
